src/Table/Column/FilterModel/TypeEnum.php

```diff
@@ -30,15 +30,16 @@ public function init(): void
 
     public function setConditionForModel($model)
     {
-        $filter = $this->recallData();
-        $values = [];
-        foreach ($filter as $key => $isSet) {
-            if ($isSet === true) {
-                $values[] = $key;
+        if ($filter = $this->recallData()) {
```

Comment:
`[]` as default?

Reply:
I was considering both options but this seems more robust I believe

Reply:
it seems fine, other checks looks like:
```

        $filter = $this->recallData();
        if (isset($filter['id'])) {
```

why can not we use the same?

Reply:
FIlterMode::recallData should definitely be declared with return type `?array`

Reply:
Maybe we do all 3, would not hurt.
And declare :array as return type.
Then it is robust :)

Reply:
if [] is good value for "not set" :) (not sure about this, probably not)

Reply:
`[]` is okay for this `isset($filter['id'])`